### PR TITLE
refactor container init

### PR DESCRIPTION
**Pull request by MosCD3**
# Summary of Changes

Replace this text with a high-level summary of the changes included in this PR.

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/app/container-imp.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export class AppContainer implements Container {
+  private container: DependencyContainer
+  public constructor(bifoldContainer: Container) {
+    this.container = bifoldContainer.getContainer().createChildContainer()
+  }
+  public init(): Container {
+    // eslint-disable-next-line no-console
+    console.log(`Initializing App container`)
+    this.container.createChildContainer().registerInstance(TOKENS.COMP_BUTTON, Button)
```
Comment (cvarjao) [inline packages/legacy/app/container-imp.tsx]: @MosCD3 , here is the issue. A new container is being created as opposed to configure the one initialized by the constructor.
replace with:
```
this.container.registerInstance(TOKENS.COMP_BUTTON, Button)
```

File: packages/legacy/app/index.js [hunk 2]
```diff
@@ -21,29 +21,31 @@ import '@formatjs/intl-datetimeformat/locale-data/en' // locale-data for en
...
 
 const navigationTheme = {
   ...NavigationTheme,
 }
 
 LogBox.ignoreAllLogs()
 // console.log(`React Native Version:${ReactNativeVersion.major}.${ReactNativeVersion.minor}.${ReactNativeVersion.patch}`)
-const MyApp = App(new MainSystem(container.createChildContainer()).init())
+const bifoldContainer = new MainContainer(container.createChildContainer()).init()
+const ONApp = BifoldApp(new AppContainer(bifoldContainer).init())
```
Comment (cvarjao) [inline packages/legacy/app/index.js]: rename `ONApp` to something else, maybe just `App`

Comment (MosCD3) [reply packages/legacy/app/index.js]: Oh sorry, forgot that

Comment (MosCD3) [reply packages/legacy/app/index.js]: done

**Review by cvarjao — CHANGES_REQUESTED** [review]
It looks good, I will remove the changes to `PodFile.lock` and `yarn.lock`. I do that last after rebase as it makes easier to rebase.

**Pull request merged**